Tasks/PerformActionOnMachineGroup/AzureHelper.ps1

```diff
@@ -71,26 +71,41 @@ function Initialize-AzureHelper
         $providerDataName = $machineGroup.ProviderDataList[0].Name
         Write-Verbose "Getting providerData : $providerDataName" -Verbose
         $providerData = Get-ProviderData -ProviderDataName $providerDataName -Connection $connection
-        $subscriptionName = $providerData.Properties.GetProperty("SubscriptionName")     
-        $username = $providerData.Properties.GetProperty("Username")
-        $password = $providerData.Properties.GetProperty("Password")
-
-        if( ![string]::IsNullOrEmpty($subscriptionName) -and ![string]::IsNullOrEmpty($username) -and ![string]::IsNullOrEmpty($password) )
+        $subscriptionId = $providerData.Properties.GetProperty("SubscriptionId")     
+        
+        if( ![string]::IsNullOrEmpty($subscriptionId) )
         {
-            Write-Verbose "SubscriptionName : $subscriptionName" -Verbose
-            Write-Verbose "Username : $username" -Verbose
-            $securePassword = ConvertTo-SecureString $password -AsPlainText -Force
-            $psCredential = New-Object System.Management.Automation.PSCredential ($username, $securePassword)
-            $azureAccount = Add-AzureAccount -Credential $psCredential
-            if(!$azureAccount)
+            $serviceEndpoint = Get-ServiceEndpoint -Name $subscriptionId -Context $distributedTaskContext
+            if ($serviceEndpoint -eq $null)
+            {
+                throw (Get-LocalizedString -Key "A Connected Service with Id '{0}' could not be found. Ensure that this Connected Service was successfully provisioned using services tab in Admin UI" -ArgumentList $subscriptionId)
```

Comment:
Where is the resource for this string? - A Connected Service with Id '{0}' could not be found. Ensure that this